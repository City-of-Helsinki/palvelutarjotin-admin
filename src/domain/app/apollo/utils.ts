import apolloClient from './apolloClient';

export const clearApolloCache = async () => {
  apolloClient.stop();
  await Promise.all([apolloClient.resetStore(), apolloClient.clearStore()]);
};
