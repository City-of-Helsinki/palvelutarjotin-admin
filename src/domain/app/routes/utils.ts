import { ROUTES } from './constants';

export const getCmsPath = (slug?: string | null) => {
  return ROUTES.CMS_PAGE.replace('/:slug', slug ?? '');
};
