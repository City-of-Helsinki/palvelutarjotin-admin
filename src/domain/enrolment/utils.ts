import { EnrolmentFieldsFragment } from '../../generated/graphql';

export const getEnrolmentFields = (
  enrolment: EnrolmentFieldsFragment | undefined | null
) =>
  enrolment
    ? {
        extraNeeds: enrolment.studyGroup.extraNeeds,
        language:
          enrolment.person?.language || enrolment.studyGroup.person.language,
        personInfo: [
          enrolment.person?.phoneNumber,
          enrolment.person?.emailAddress,
        ].filter((e) => e),
        studyGroupPersonInfo: [
          enrolment.studyGroup.person.phoneNumber,
          enrolment.studyGroup.person.emailAddress,
        ].filter((e) => e),
      }
    : {};
