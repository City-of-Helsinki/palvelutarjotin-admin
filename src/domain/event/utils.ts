import { SUPPORT_LANGUAGES } from '../../constants';
import { EventQuery } from '../../generated/graphql';
import { Language } from '../../types';
import { EVENT_PLACEHOLDER_IMAGES } from './constants';
/**
 * Get event placeholder image url
 * @param {string} id
 * @return {string}
 */
export const getEventPlaceholderImage = (id: string): string => {
  const numbers = id.match(/\d+/g);
  const sum = numbers
    ? numbers.reduce((prev: number, cur: string) => prev + Number(cur), 0)
    : 0;
  const index = sum % 4;

  return EVENT_PLACEHOLDER_IMAGES[index];
};

/**
 * Get first availanle language of an event
 * @param {object} eventData
 * @return {string}
 */
export const getFirstAvailableLanguage = (eventData: EventQuery): Language => {
  if (eventData.event?.name.en) return SUPPORT_LANGUAGES.EN;
  if (eventData.event?.name.fi) return SUPPORT_LANGUAGES.FI;
  if (eventData.event?.name.sv) return SUPPORT_LANGUAGES.SV;

  return SUPPORT_LANGUAGES.FI;
};
