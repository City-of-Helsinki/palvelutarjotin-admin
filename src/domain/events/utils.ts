import { isFuture } from 'date-fns';
import forEach from 'lodash/forEach';

import { EventFieldsFragment } from '../../generated/graphql';

export const hasOccurrences = (event: EventFieldsFragment) => {
  return Boolean(event.pEvent?.occurrences.edges.length);
};

export const hasComingOccurrences = (event: EventFieldsFragment) => {
  let hasComingItems = false;

  forEach(event.pEvent?.occurrences.edges, (edge) => {
    if (edge?.node?.startTime && isFuture(new Date(edge?.node?.startTime))) {
      hasComingItems = true;
      return false;
    }
  });

  return Boolean(hasComingItems);
};
