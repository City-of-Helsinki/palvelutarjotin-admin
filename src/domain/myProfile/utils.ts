import {
  OrganisationNodeFieldsFragment,
  PersonFieldsFragment,
} from '../../generated/graphql';
import { MyProfileFormFields } from './myProfileForm/MyProfileForm';

export const getMyProfilePayload = (values: MyProfileFormFields) => ({
  name: values.name,
  phoneNumber: values.phoneNumber,
  emailAddress: values.emailAddress,
  organisations: values.organisations,
});

export const getSelectedOrganisation = (
  myProfile: PersonFieldsFragment,
  activeOrganisation: string | null
): OrganisationNodeFieldsFragment | null => {
  const organisations =
    myProfile.organisations.edges.map((edge) => ({
      ...(edge?.node as OrganisationNodeFieldsFragment),
    })) || [];

  const organisation = activeOrganisation
    ? organisations.find((item) => item.id === activeOrganisation)
    : null;

  return organisation || organisations[0];
};
