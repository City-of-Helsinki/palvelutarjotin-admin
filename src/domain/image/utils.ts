import { ImageDocument, ImageQuery } from '../../generated/graphql';
import { initializeApolloClient } from '../app/apollo/apolloClient';

export const getImageName = (id: string) => {
  const apolloClient = initializeApolloClient();
  const data = apolloClient.readQuery<ImageQuery>({
    query: ImageDocument,
    variables: { id },
  });
  return data?.image?.name;
};
