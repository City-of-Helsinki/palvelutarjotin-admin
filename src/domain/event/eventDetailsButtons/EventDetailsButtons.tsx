import { Button } from 'hds-react';
import * as React from 'react';
import { useTranslation } from 'react-i18next';

import BackButton from '../../../common/components/backButton/BackButton';
import { SUPPORT_LANGUAGES } from '../../../constants';
import { EventQuery } from '../../../generated/graphql';
import useGoBack from '../../../hooks/useGoBack';
import useHistory from '../../../hooks/useHistory';
import { Language } from '../../../types';
import useQueryStringWithReturnPath from '../../../utils/useQueryStringWithReturnPath';
import { ROUTES } from '../../app/routes/constants';
import EventLanguageSelector from '../eventLanguageSelector/EventLanguageSelector';
import { isEditableEvent } from '../utils';
import styles from './eventDetailsButtons.module.scss';

interface Props {
  eventData: EventQuery;
  onClickLanguage: (language: Language) => void;
  selectedLanguage: Language;
}

const EventDetailsButtons: React.FC<Props> = ({
  eventData,
  onClickLanguage,
  selectedLanguage,
}) => {
  const { t } = useTranslation();
  const history = useHistory();
  const eventId = eventData.event?.id || '';
  const goBack = useGoBack({
    defaultReturnPath: ROUTES.HOME,
    pageQueryParams: ['language'],
  });
  const queryStringWithReturnPath = useQueryStringWithReturnPath();

  const goToEditPage = () => {
    const searchParams = new URLSearchParams(queryStringWithReturnPath);
    searchParams.append('language', selectedLanguage);
    history.pushWithLocale({
      pathname: ROUTES.EDIT_EVENT.replace(':id', eventId),
      search: searchParams.toString(),
    });
  };

  return (
    <div className={styles.eventDetailsButtons}>
      <div>
        <BackButton onClick={goBack}>{t('common.back')}</BackButton>
      </div>
      <EventLanguageSelector
        languages={Object.values(SUPPORT_LANGUAGES).map((language) => ({
          isCompleted: !!eventData.event?.name[language],
          isDisabled: !eventData.event?.name[language],
          label: t(`common.languages.${language}`),
          value: language,
        }))}
        onClick={onClickLanguage}
        selectedLanguage={selectedLanguage}
      />
      {isEditableEvent(eventData) && (
        <div>
          <Button onClick={goToEditPage}>
            {t('eventDetails.buttons.buttonEdit')}
          </Button>
        </div>
      )}
    </div>
  );
};

export default EventDetailsButtons;
