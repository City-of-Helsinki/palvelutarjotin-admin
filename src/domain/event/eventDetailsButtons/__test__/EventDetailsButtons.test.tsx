import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import * as React from 'react';
import * as Router from 'react-router-dom';
import { vi } from 'vitest';

import { fakeEvent } from '../../../../utils/mockDataUtils';
import { customRender } from '../../../../utils/testUtils';
import EventDetailsButtons from '../EventDetailsButtons';

const navigate = vi.fn();
vi.mock('react-router-dom', async () => {
  const actual = await vi.importActual('react-router-dom');
  return { ...actual };
});
const eventId = 'palvelutarjotin:afzunowba4';

const event = fakeEvent({
  id: eventId,
});

afterAll(() => {
  vi.setSystemTime(vi.getRealSystemTime());
  vi.useRealTimers();
});

afterEach(() => {
  vi.restoreAllMocks();
});

// test('is accessible and matches snapshot', async () => {
//   const { container } = render(
//     <EventDetailsButtons
//       eventData={{ event }}
//       onClickLanguage={vi.fn()}
//       selectedLanguage="fi"
//     />
//   );

//   const result = await axe(container);

//   expect(result).toHaveNoViolations();
//   expect(container).toMatchSnapshot();
// });

test('it renders correct texts and click events work', async () => {
  vi.spyOn(Router, 'useNavigate').mockImplementation(() => navigate);
  vi.setSystemTime(new Date(2020, 6, 10));

  customRender(
    <EventDetailsButtons
      eventData={{ event }}
      onClickLanguage={vi.fn()}
      selectedLanguage="fi"
    />
  );

  const backButton = screen.getByRole('button', { name: 'Takaisin' });
  await userEvent.click(backButton);

  expect(navigate).toHaveBeenCalledTimes(1);
  expect(navigate).toHaveBeenCalledWith(
    {
      pathname: '/fi',
      search: '',
    },
    expect.anything()
  );

  const editButton = screen.getByRole('button', { name: 'Muokkaa tapahtumaa' });
  await userEvent.click(editButton);

  expect(navigate).toHaveBeenCalledTimes(2);
  expect(navigate).toHaveBeenCalledWith(
    {
      pathname: '/fi',
      search: '',
    },
    expect.anything()
  );

  expect(screen.queryByText('Takaisin')).toBeVisible();
  expect(screen.queryByText('suomi')).toBeVisible();
  expect(screen.queryByText('englanti')).toBeVisible();
  expect(screen.queryByText('ruotsi')).toBeVisible();
});
