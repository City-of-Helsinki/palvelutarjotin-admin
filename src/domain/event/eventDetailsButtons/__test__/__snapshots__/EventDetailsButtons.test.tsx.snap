// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is accessible and matches snapshot 1`] = `
<div>
  <div
    class="eventDetailsButtons"
  >
    <div>
      <button
        class="backButton"
      >
        <div
          class="backIconWrapper"
        >
          <svg
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M10 5.636l1.414 1.414L7.465 11H20v2H7.464l3.95 3.95L10 18.364 3.636 12z"
                fill="currentColor"
              />
            </g>
          </svg>
        </div>
        <span>
          Tapahtumat
        </span>
      </button>
    </div>
    <div
      class="eventLanguageSelector"
    >
      <button
        class="languageButton isSelected"
        type="button"
      >
        <div
          class="buttonIconWrapper isCompleted"
        >
          <svg
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M20.763 7.414L10 18l-5.5-5.5 1.25-1.25 4.25 4L19.35 6z"
                fill="currentColor"
              />
            </g>
          </svg>
        </div>
        <span>
          Suomi
        </span>
      </button>
      <button
        class="languageButton isDisabled"
        disabled=""
        type="button"
      >
        <div
          class="buttonIconWrapper"
        >
          <svg
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm1 6v6.5h2V18H9v-1.5h2v-5H9V10h4zm-1.187-4a1.312 1.312 0 110 2.625 1.312 1.312 0 010-2.625z"
                fill="currentColor"
              />
            </g>
          </svg>
        </div>
        <span>
          Ruotsi
        </span>
      </button>
      <button
        class="languageButton isDisabled"
        disabled=""
        type="button"
      >
        <div
          class="buttonIconWrapper"
        >
          <svg
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm1 6v6.5h2V18H9v-1.5h2v-5H9V10h4zm-1.187-4a1.312 1.312 0 110 2.625 1.312 1.312 0 010-2.625z"
                fill="currentColor"
              />
            </g>
          </svg>
        </div>
        <span>
          Englanti
        </span>
      </button>
    </div>
  </div>
</div>
`;
